tests/modeltests/tst_models.cpp

```diff
@@ -537,7 +537,7 @@ private slots:
         model.setData(events);
 
         QCOMPARE(model.columnCount(), static_cast<int>(EventModel::NUM_COLUMNS));
-        QCOMPARE(model.rowCount(), 2);
+        QCOMPARE(model.rowCount(), 4);
```

Comment:
what are the new rows? 

please also add test coverage for the new favorite feature. we also need to be able to un-favorite a row

Reply:
Favourites and tracepoints

Reply:
but without a favorite, there should be no additional (empty) row. similarly, without any tracepoints, there shouldn't be an empty row for that either

Reply:
I added a proxy to hide the empty rows in the timelinewidget